### ISSUE TITLE
Using EvaporateJS in Internet Explorer

### ISSUE THREAD

**Issue by thomaslange24** [TTLabs/EvaporateJS#436]
Hi
I need to use Evaporate in the Internet Explorer and since IE does not support Promises i will have to go with an older version of EvaporateJS, like v1.6.4. Right?
Now i tried the javascript/html example with aws signature version 4. But i am receiving a status code 403 Forbidden for the uploading although i am using root credentials. (i am working with a minio storage). What could i do about that?

**Comment by jakubzitny** [comment]
First of all, you can use latest version, you just have to provide an implementation of Promises somehow. 

And second of all, without repro code, there is not much we can do, right? :P

**Comment by thomaslange24** [comment]
> you can use latest version, you just have to provide an implementation of Promises somehow

That makes sense. Now i understand. I am pretty new to javascript and web applications. So i was not aware. I think in that case i will definitely go with the latest version.

So on https://github.com/TTLabs/EvaporateJS/wiki/Browser-Compatibility there is recommendation for the promise polyfill https://github.com/stefanpenner/es6-promise. Does that mean i just have to include this in my html file to make Evaporate work in IE?

**Comment by thomaslange24** [comment]
Yes, it works by just adding the corresponding source tag.
<script src="https://cdn.jsdelivr.net/npm/es6-promise@4/dist/es6-promise.auto.js"></script>
But it does not work with the minified version.

**Comment by thomaslange24** [comment]
Well, i was wrong. The code gets executed and the requests are send. But the upload fails when completing the multipart upload there is a response with status 400 and the following:

> complete error: testbucket/uploadfile1MB.bin AWS Code: InvalidPart, Message:One or more of the 
specified parts could not be found.  The part may not have been uploaded, or the specified entity tag may not match the part&#39;s entity tag.status:400

This happens in IE11 but not in Google Chrome. What can i do?
[evaporate_example.zip](https://github.com/TTLabs/EvaporateJS/files/3329677/evaporate_example.zip)

**Comment by thomaslange24** [comment]
I am using a minio server for that.

**Comment by jakubzitny** [comment]
1. We've never tried minio for this, can you try the same code with s3?

2. Does this work in a different browser, e.g. Chrome?

**Comment by thomaslange24** [comment]
Minio is S3 compatible and you can use the aws sdk with minio. Also this example works in Chrome.

**Comment by thomaslange24** [comment]
Setting up a minio server is easy and pretty fast: https://docs.min.io/

**Comment by thomaslange24** [comment]
I need to make this example work with minio and in the IE11

**Comment by thomaslange24** [comment]
But sunscreen it works in Chrome I would guess it has something to do with IE. Not with Minio.

**Comment by jakubzitny** [comment]
How about Edge? Also, did you look at all the logs in Console and Requests being sent from IE?

**Comment by thomaslange24** [comment]
It does not work in edge either. But that looks different. It seems that the upload is not even started in edge. I am a total beginner in working with the browser. So i might have missed something, but i am not aware. Maybe it helps if you see the logs?
[logs.zip](https://github.com/TTLabs/EvaporateJS/files/3344579/logs.zip)

(With the credentials for minio play you could try the example youself without starting a minio server yourself. But it looks like this is not available at the moment.)

**Comment by thomaslange24** [comment]
Although when i start the multipart upload in IE for a file the first time it fails. But when i reload the page and try the same upload again it completes. At least for a small file.

**Comment by thomaslange24** [comment]
I added the console output for IE and Edge:
[logs.zip](https://github.com/TTLabs/EvaporateJS/files/3344688/logs.zip)

**Comment by thomaslange24** [comment]
Does this help?

**Comment by jakubzitny** [comment]
Not really, but at one point it says "all files were uploaded successfully". Can you see anything like that in the Minio logs?

Also here https://github.com/aws-amplify/aws-sdk-android/issues/309 they noticed similar message happens when the upload pauses after the start. And based on the [docs](https://docs.aws.amazon.com/AmazonS3/latest/API/mpUploadComplete.html) I'd say that it happens when requesting the multipart-upload completion (which is the last POST request on multipart upload). I'd try investigating what's actually sent from the browser to the storage.

In the edge har log I can't see the actual request, could you please send a new one?

**Comment by thomaslange24** [comment]
The "all files were uploaded successfully" happened after i reloaded the page when the upload was not succesful. But then i deleted the object from the storage and tried the upload againfrom Internet Explorer. The second time it failed again just like before.

The stuff you see in the edge har log is all that happens. It won't start correctly.

Right now i can not redo the case. I reinstalled evaporate and now it ignores my aws url and just uses s3.amazonaws.com. That is with the same configuration like before.
